x-pack/plugins/actions/server/lib/get_token_count_from_invoke_stream.ts

```diff
@@ -195,3 +211,31 @@ const parseOpenAIResponse = (responseBody: string) =>
       prev += msg.content || '';
       return prev;
     }, '');
+
+/**
+ * Parses the final chunk of a Bedrock buffer to extract the usage object.
+ * @param finalChunk
+ */
+const getUsageFromFinalChunk = (
+  finalChunk: Uint8Array
+): {
+  total: number;
+  prompt: number;
+  completion: number;
+} | null => {
+  const awsDecoder = new EventStreamCodec(toUtf8, fromUtf8);
+  const event = awsDecoder.decode(finalChunk);
+  const body = JSON.parse(
+    Buffer.from(JSON.parse(new TextDecoder().decode(event.body)).bytes, 'base64').toString()
+  );
+  if (body.type === 'message_stop') {
+    return {
+      total:
+        body['amazon-bedrock-invocationMetrics'].inputTokenCount +
```

Comment:
I'm a little confused why this doesn't generate type warnings about referencing fields in an `any`.  Another case where validating these before hand, with an error log if not the expected shape, will help if we run into issues.